creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -555,83 +408,35 @@ class SnapbackSM {
-        throw new Error('processStateMachineOperation():getAllNodeUsers() Error')
-      }
-
-      /**
-       * Select chunk of users to process in this run
-       *  - User is selected if (user_id % ModuloBase = currentModuloSlice)
-       *  - nodeUsers = array of objects of schema { primary, secondary1, secondary2, user_id, wallet }
-       */
-      nodeUsers = nodeUsers.filter(
-        nodeUser => (nodeUser.user_id % ModuloBase === this.currentModuloSlice)
-      )
-      decisionTree.push({
-        stage: '3/ select nodeUsers modulo slice',
-        vals: { nodeUsersSubsetLength: nodeUsers.length, moduloBase: ModuloBase, currentModuloSlice: this.currentModuloSlice },
-        time: Date.now()
-      })
-
-      // Compute content node peerset from nodeUsers (all nodes that are in a shared replica set with this node)
-      let peerSet = []
-      try {
-        peerSet = this.computeContentNodePeerSet(nodeUsers)
+      nodeUsers = await this.peerSetManager.getNodeUsers()
+      nodeUsers = this.peerSetManager.sliceUsers(nodeUsers)
 
-        decisionTree.push({
-          stage: '4/ computeContentNodePeerSet() Success',
-          vals: { peerSetLength: peerSet.size },
-          time: Date.now()
-        })
-      } catch (e) {
-        decisionTree.push({ stage: '4/ computeContentNodePeerSet() Error', vals: e.message, time: Date.now() })
-        throw new Error('processStateMachineOperation():computeContentNodePeerSet() Error')
-      }
-
-      /**
-       * Determine health for every peer & build list of unhealthy peers
-       * TODO change from sequential to chunked parallel
-       */
-      const unhealthyPeers = new Set()
-      try {
-        for await (const peer of peerSet) {
-          const peerIsHealthy = await this.determinePeerHealth(peer)
-          if (!peerIsHealthy) {
-            unhealthyPeers.add(peer)
-          }
-        }
+      decisionTree.push({ stage: 'getNodeUsers() and sliceUsers() Success', vals: { nodeUsersLength: nodeUsers.length }, time: Date.now() })
+    } catch (e) {
+      decisionTree.push({ stage: 'getNodeUsers() or sliceUsers() Error', vals: e.message, time: Date.now() })
+      throw new Error('processStateMachineOperation():getNodeUsers()/sliceUsers() Error')
+    }
 
-        decisionTree.push({
-          stage: '5/ determinePeerHealth() Success',
-          vals: {
-            unhealthyPeerSetLength: unhealthyPeers.size,
-            healthyPeerSetLength: peerSet.size - unhealthyPeers.size,
-            unhealthyPeers: Array.from(unhealthyPeers)
-          },
-          time: Date.now()
-        })
-      } catch (e) {
-        decisionTree.push({ stage: '5/ determinePeerHealth() Error', vals: e.message, time: Date.now() })
-        throw new Error('processStateMachineOperation():determinePeerHealth() Error')
-      }
+    let unhealthyPeers = await this.peerSetManager.getUnhealthyPeers(nodeUsers)
```

Comment:
This should be in its own try catch

Reply:
discussed offline -- outer try/catch block will append to decisionTree if necessary. resolving